sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/RewriteWithExpressionSuite.scala

```diff
@@ -29,7 +28,7 @@ import org.apache.spark.sql.catalyst.rules.RuleExecutor
 class RewriteWithExpressionSuite extends PlanTest {
 
   object Optimizer extends RuleExecutor[LogicalPlan] {
-    val batches = Batch("Rewrite With expression", Once,
+    val batches = Batch("Rewrite With expression", FixedPoint(5),
```

Comment:
Do you support any levels of nested `with` or there is some restrictions? 5 is enough in any case?

Reply:
The implementation supports any nesting levels but we only test 2 levels here so `FixedPoint(5)` is sufficient